### ISSUE TITLE
retroarch Playstation 2 freezes after a few minutes of playing gba 

### ISSUE THREAD

**Issue by andres0021** [libretro/RetroArch#15763]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
After a few minutes of playing, PS2 retroarch freezes, I have to restart the console, this happens to me while I'm playing GBA, I notice that the crash happened to me on the PS2 console, on the PS2 emulator I don't know if it crashes. 

[Description of the bug]

### Expected behavior

[What you expected to happen]

### Actual behavior

[What is actually happening]

### Steps to reproduce the bug

1. [First step]
2. [Second step]
3. [and so on...]

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]

### Environment information

- OS: [The operating system you're running]
- Compiler: [In case you are running local builds]

**Comment by zoltanvb** [comment]
Can you please tell which core (gpSP or mgba), and what rom is used, when freezing occurs?

**Comment by andres0021** [comment]
> Can you please tell which core (gpSP or mgba), and what rom is used, when freezing occurs?

gpSP, and it freezes with several games, for example sonic advance 1

**Comment by zoltanvb** [comment]
I could not reproduce this with current version. Starting from today (10-25), the nightly build should have sound again, can you test with that one?
http://buildbot.libretro.com/nightly/playstation/ps2/

**Comment by andres0021** [comment]
> I could not reproduce this with current version. Starting from today (10-25), the nightly build should have sound again, can you test with that one? http://buildbot.libretro.com/nightly/playstation/ps2/

Hello, I just tried the one from today the 25th, and now it has sound. One question, do you need a PS2 tester? I say this so that you can repair the gpSP crashes, I am willing to help in case you need help with that

**Comment by zoltanvb** [comment]
Yes, please check if the gpSP crash still occurs with the current version. This core was in active development recently, so there may be changes.

**Comment by andres0021** [comment]
> Yes, please check if the gpSP crash still occurs with the current version. This core was in active development recently, so there may be changes.

Well yesterday I tried the version dated 25, and it still has freezes, I hope one day they can fix it :)

**Comment by zoltanvb** [comment]
Can you try generating a log? It is in Settings/Logging, you can enable all items there and set level to 0 for both core and frontend. After turning it on, play until freeze occurs. Logs can be found under retroarch/logs.

**Comment by andres0021** [comment]
> Can you try generating a log? It is in Settings/Logging, you can enable all items there and set level to 0 for both core and frontend. After turning it on, play until freeze occurs. Logs can be found under retroarch/logs.

Hello, I just did those steps that you mentioned, but when I want to copy the log files they come out 0 kb, they are damaged in some way I think

**Comment by andres0021** [comment]
> Can you try generating a log? It is in Settings/Logging, you can enable all items there and set level to 0 for both core and frontend. After turning it on, play until freeze occurs. Logs can be found under retroarch/logs.

I couldn't, I mean when I activate those options, when I try to open a game it always automatically sends me to the beginning of the memory card

**Comment by andres0021** [comment]
> Can you try generating a log? It is in Settings/Logging, you can enable all items there and set level to 0 for both core and frontend. After turning it on, play until freeze occurs. Logs can be found under retroarch/logs.

Hello, I just tried the ee update today the 30th, and now it does let me do the log thing, but it leaves me with files all at 0 kb, also again the retroarch was left without sound

**Comment by zoltanvb** [comment]
0 byte log does occur for me sometimes too, usually when core freezes for some reason, I have not yet found why.

I have also tried the build from 2023-10-23, but seems to work here, both sound and gpsp. Do you overwrite all the cores, when you extract the new build, and do you keep your old retroarch.cfg file? If so, can you share the .cfg?

**Comment by andres0021** [comment]
> 0 byte log does occur for me sometimes too, usually when core freezes for some reason, I have not yet found why.
> 
> I have also tried the build from 2023-10-23, but seems to work here, both sound and gpsp. Do you overwrite all the cores, when you extract the new build, and do you keep your old retroarch.cfg file? If so, can you share the .cfg?

I didn't understand very well, but normally when I try a new update I delete the entire "retroarch" folder and paste a new one on my USB.

**Comment by andres0021** [comment]
> 0 byte log does occur for me sometimes too, usually when core freezes for some reason, I have not yet found why.
> 
> I have also tried the build from 2023-10-23, but seems to work here, both sound and gpsp. Do you overwrite all the cores, when you extract the new build, and do you keep your old retroarch.cfg file? If so, can you share the .cfg?

Hey, for some reason in today's update 31 I can't load the roms, I'm deciding to open a game but I can't sell the file to start a rom

**Comment by zoltanvb** [comment]
The reason why I am asking is that it looks as if older cores get loaded somehow, and retroarch.cfg contains the core path.
I am on discord, libretro server, same name.

**Comment by andres0021** [comment]
> The reason why I am asking is that it looks as if older cores get loaded somehow, and retroarch.cfg contains the core path. I am on discord, libretro server, same name.

I understand, so I'll give you retroarch.cfg here 
https://www.mediafire.com/file/2v4hny8x1zsg6m8/retroarch.cfg/file
